src/common/ColourPalette.hpp

```diff
@@ -31,15 +33,53 @@ class ColourPalette {
         /** Converts a given palette value into packed RGB (format 0x00RRGGBB). */
         uInt32 getRGB(int val) const;
 
-    private:
+        /** returns a pointer to the palette array (256 elements) */
+        const uInt32 *getPalette();
+
+        /**
+          Loads all defined palettes with PAL color-loss data depending
+          on 'state'.
+          Sets the palette according to the given palette name.
+
+          @param type  The palette type = {standard, z26, user}
+          @param displayFormat The display format = { NTSC, PAL, SECAM }
+        */
+        void setPalette(const string& type,
+                        const string& displayFormat);
+
+        /**
+            Loads a user-defined palette file (from OSystem::paletteFile), filling the
+            appropriate user-defined palette arrays.
+        */
+        void loadUserPalette(const string& paletteFile);
+
+private:
+
+        /**
+         *  Calculates grayscale values for all palettes
+         */
+        void calculateGrayscaleValues();
+
+        uInt32 *m_palette;
+
+        bool myUserPaletteDefined;
+
+        // Table of RGB values for NTSC, PAL and SECAM
+        static uInt32 NTSCPalette[256];
```

Comment:
One suggestion: I would move these to the cpp file (they don't need to be member variables of ColourPalette) so that they don't clutter the ColourPalette interface.